packages/api/app/Models/Technology.js

```diff
@@ -121,6 +121,24 @@ class Technology extends Model {
 			.wherePivot('role', 'OWNER')
 			.first();
 	}
+
+	async checkResponsible(user, role = null) {
+		let responsible;
+		if (role) {
+			responsible = await this.users()
+				.where({ user_id: user.id })
+				.wherePivot('role', role)
+				.first();
+		} else {
+			responsible = await this.users()
+				.where({ user_id: user.id })
+				.first();
+		}
+		if (responsible) {
+			return true;
+		}
+		return false;
```

Comment:
```suggestion
		let responsible = this.users().select('id').where({ user_id: user.id })

		if (role) {
			responsible = responsible.wherePivot('role', role);
		}

		responsible = await responsible.first();

		return !!responsible;
```

Teste antes só pra garantir :slightly_smiling_face:

Reply:
Deu certo :+1: